src/app/shared/autocomplete/autocomplete.component.ts

```diff
@@ -0,0 +1,86 @@
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from '@angular/forms';
+import {Subscription} from 'rxjs';
+import * as citiesJsonData from '../constants/cities.constant.json';
+
+export interface City {
+  country: string;
+  geonameid: number;
+  name: string;
+  subcountry: string;
+}
+
+@Component({
+  selector: 'app-autocomplete',
+  templateUrl: './autocomplete.component.html',
+  styleUrls: ['./autocomplete.component.less']
+})
+export class AutocompleteComponent implements OnInit, OnDestroy {
```

Comment:
In common reusable components, we should avoid straight naming and use a common naming